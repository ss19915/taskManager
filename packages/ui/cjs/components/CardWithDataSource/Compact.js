"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _theme = require("@react-web-app/theme");

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Compact = function Compact(props) {
  var children = props.children,
      compact = props.compact,
      label = props.label;

  if (compact) {
    return children;
  } else if (label) {
    return _react["default"].createElement(_theme.Paper, {
      basic: true
    }, _react["default"].createElement(_theme.Paper, null, label, _react["default"].createElement(_theme.Paper, {
      basic: true
    }, children)));
  }

  return _react["default"].createElement(_theme.Paper, {
    basic: true
  }, _react["default"].createElement(_theme.Paper, null, children));
};

Compact.propTypes = {
  children: _propTypes["default"].node,
  compact: _propTypes["default"].bool,
  label: _propTypes["default"].node
};
var _default = Compact;
exports["default"] = _default;