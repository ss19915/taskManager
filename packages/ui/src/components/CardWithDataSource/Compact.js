import React from 'react';
import { Paper } from '@react-web-app/theme';
import T from 'prop-types';

const Compact = (props) => {
    const {
        children,
        compact,
        label
    } = props;

    if (compact) {
        return (children);
    }

    else if (label) {
        return (
            <Paper basic>
                <Paper>
                    {label}
                    <Paper basic>
                        {children}
                    </Paper>
                </Paper>
            </Paper>
        );
    }

    return (
        <Paper basic>
            <Paper>
                {children}
            </Paper>
        </Paper>
    );
};

Compact.propTypes = {
    children: T.node,
    compact: T.bool,
    label: T.node,
};

export default Compact;
