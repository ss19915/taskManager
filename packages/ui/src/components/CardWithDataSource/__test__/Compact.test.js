import React from 'react';
import Compact from '../Compact';

describe('Component: Compact', () => {
    it('Should render correctly with no child', () => {
        const wrapper = shallow(<Compact />);

        expect(wrapper).toMatchSnapshot();
    });

    it('Should render correctly with childs', () => {
        const A = () => (<p>A</p>);

        const wrapper = shallow(<Compact><A /></Compact>);

        expect(wrapper).toMatchSnapshot();
    });

    it('Should be compack when prop compack is passed', () => {
        const A = () => (
            <div>
                B
                <p>A</p>
            </div>
        );

        const wrapper = render(
            <Compact compact>
                <A />
            </Compact>
        );

        expect(wrapper.html()).toEqual(render(<A />).html());
    });
});