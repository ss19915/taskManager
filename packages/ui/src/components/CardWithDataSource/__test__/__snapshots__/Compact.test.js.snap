// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Compact Should render correctly with childs 1`] = `
<WithStyles(ForwardRef(Paper))
  basic={true}
>
  <WithStyles(ForwardRef(Paper))>
    <A />
  </WithStyles(ForwardRef(Paper))>
</WithStyles(ForwardRef(Paper))>
`;

exports[`Component: Compact Should render correctly with no child 1`] = `
<WithStyles(ForwardRef(Paper))
  basic={true}
>
  <WithStyles(ForwardRef(Paper)) />
</WithStyles(ForwardRef(Paper))>
`;
